bokehjs/src/lib/models/widgets/file_input.ts

```diff
@@ -28,19 +28,30 @@ export class FileInputView extends WidgetView {
 
   load_file(e: any): void {
     const reader = new FileReader()
+    this.model.filename = e.target.files[0].name
     reader.onload = (e) => this.file(e)
     reader.readAsDataURL(e.target.files[0])
   }
 
   file(e: any): void {
```

Comment:
If you are at it, you could also improve those `any` types. This one should be replaced with `ProgressEvent`. You can learn precise types from https://github.com/microsoft/TypeScript/blob/master/lib/lib.dom.d.ts (interface `FileReader`).